src/server.c

```diff
@@ -6756,7 +6763,10 @@ int main(int argc, char **argv) {
 #endif
     tzset(); /* Populates 'timezone' global. */
     zmalloc_set_oom_handler(serverOutOfMemoryHandler);
-
+#if defined(USE_JEMALLOC)
+    // we assume jemalloc version in use supports defragmentation api
```

Comment:
I removed also the `JEMALLOC_FRAG_HINT` flag from [here](https://github.com/valkey-io/valkey/blob/38457b73208d9edadbdeb64dda8b18f57099475a/deps/jemalloc/include/jemalloc/jemalloc_macros.h.in#L153) because now we move to use jemalloc native API, we can expect it to be supported if we use jemalloc -> so HAVE_DEFRAG is true if JEMALLOC. But, I can do something like what we have in[ zmalloc.h ](https://github.com/valkey-io/valkey/blob/38457b73208d9edadbdeb64dda8b18f57099475a/src/zmalloc.h#L50) to make sure it's supported. Do you think it's better?

Reply:
Is there a mininum required jemalloc version then?

Reply:
Minimum required is **5.2.1** see [here](https://github.com/jemalloc/jemalloc/commit/9aab3f2be041b09f42375d3bf173d1a8795a1ee9)

Reply:
OK, then we should probably check these macros (https://github.com/jemalloc/jemalloc/blob/dev/include/jemalloc/jemalloc_macros.h.in) with `#if` before we enable defrag, right?

```
#define JEMALLOC_VERSION "@jemalloc_version@"
#define JEMALLOC_VERSION_MAJOR @jemalloc_version_major@
#define JEMALLOC_VERSION_MINOR @jemalloc_version_minor@
#define JEMALLOC_VERSION_BUGFIX @jemalloc_version_bugfix@
```

Otherwise, what happens if the user compiles with an older jemalloc?

Reply:
Or are you saying that 5.2.1 is already so old that all users and distros already include it? Then, it's all fine. :heavy_check_mark: 

The oldest ones we have in our CI is CentOS Stream 9, AlmaLinux 8, RHEL 8, right? They all seem to include 5.2.1 according to https://pkgs.org/download/jemalloc.

Reply:
> But, I can do something like what we have in[ zmalloc.h ](https://github.com/valkey-io/valkey/blob/38457b73208d9edadbdeb64dda8b18f57099475a/src/zmalloc.h#L50) to make sure it's supported. Do you think it's better?
@zuiderkwast, good point, agree we should check the version. I can follow this existing version check, to make support use cases where someone want's to build valkey with older jemalloc version or link to pre-build lib

Reply:
@madolson you are correct, to support older than 5.2.1 jemalloc versions, I should introduce version check(but no need to check the hint, just the version)